### PR TITLE
Retry in the find function for fs.realpathSync method for UNC-paths

### PR DESCRIPTION
**Pull request by DenisRumyantsev**
**Description**: occasionally the `find` function fails with the error `Failed find: EISDIR: illegal operation on a directory`. Most likely the reason is that the `fs.realpathSync` method works unstable for UNC-paths. So we should add retry logic for this method for UNC-paths.

**Changelog**:
* `node/task.ts` — the `find` function — implemented retry for the `fs.realpathSync` method for UNC-paths using `isUncPath` and `retry` functions.
* `node/test/isuncpathtests.ts` — implemented unit-test for the `isUncPath` function.
* `node/test/retrytests.ts` — implemented unit-test for the `retry` function.
* `node/mock-task.ts` — implemented mocks for `isUncPath` and `retry` functions.

[Related issue](https://github.com/microsoft/azure-pipelines-tasks/issues/14789)

### CODE REVIEW

**Review by EzzhevNikita — APPROVED** [review]

File: node/task.ts [hunk 1]
```diff
@@ -911,7 +967,12 @@ export function find(findPath: string, options?: FindOptions): string[] {
 
                 if (options.followSymbolicLinks) {
                     // get the realpath
-                    let realPath: string = fs.realpathSync(item.path);
+                    let realPath: string;
+                    if (isUncPath(item.path)) {
+                        realPath = retry(fs.realpathSync, [item.path], { continueOnError: false, retryCount: 5 });
+                    } else {
+                        realPath = fs.realpathSync(item.path);
+                    }
```
Comment (alexander-smolyakov) [inline node/task.ts:975]: Could you please write some comments in the code regarding why we retrying if the provided path is UNC?

File: node/task.ts [hunk 2]
```diff
@@ -812,6 +812,62 @@ export interface FindOptions {
...
 
+/**
+ * Defines if path is unc-path.
+ *
+ * @param path  a path to a file.
+ * @returns     true if path starts with double backslash, otherwise returns false.
+ */
+export function isUncPath(path: string) {
+    return /^\\\\[^\\]/.test(path);
+}
```
Comment (alexander-smolyakov) [inline node/task.ts:823]: Could it make sense to move this function to [internal.ts](https://github.com/microsoft/azure-pipelines-task-lib/blob/master/node/internal.ts)?

**Review by alexander-smolyakov — COMMENTED** [review]
@DenisRumyantsev could you please take a look at the comments?

**Comment by alexander-smolyakov** [comment]
@DenisRumyantsev, it seems CI checks are failed, could you please take a look at them as well?

**Review by alexander-smolyakov — APPROVED** [review]

**Review by DaniilShmelev — APPROVED** [review]

**Review by EzzhevNikita — APPROVED** [review]

**Pull request merged**